### PR TITLE
Make sure redirects to a Location without a port work

### PR DESCRIPTION
**Pull request by encukou**
The redirect handling code raised errors when redirected to an URL without a port number set.
This adds an `add_port` function that adds a default port if missing, and calls it in the redirect handling code.

### CODE REVIEW

File: tests/test_add_port.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import pytest
+
+from freezeyt.util import parse_absolute_url, add_port
+
+url_with_port = parse_absolute_url('http://localhost:80/')
+
+def test_add_http_port():
+    url_without_port = url_with_port.join('http://localhost/')
```
Comment (nappex) [inline tests/test_add_port.py:8]: Why we don't use just `parse_absolute_url('http://localhost/')`

File: tests/test_add_port.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    assert add_port(url_without_port).to_url() == 'https://localhost:443/'
+
+def test_add_port_unknown_scheme():
+    url_without_port = url_with_port.join('unknownscheme://localhost/')
+    assert url_without_port.to_url() == 'unknownscheme://localhost/'
+    with pytest.raises(ValueError):
+        add_port(url_without_port)
+
+def test_no_add_http_port():
+    url_without_port = url_with_port.join('http://localhost:1234/')
```
Comment (nappex) [inline tests/test_add_port.py:24]: Why have we named var as `url_without_port` when we save to it URL with port ?

Comment (encukou) [reply tests/test_add_port.py:24]: That's a typo, need to fix that.

File: tests/test_add_port.py (hunk 1 above)
Comment (encukou) [reply tests/test_add_port.py:8]: Our `parse_absolute_url` always adds the port :)

**Review by nappex — APPROVED** [review]

**Pull request merged**